client/state/jetpack/connection/reducer.js

```diff
@@ -45,6 +49,32 @@ export const requests = createReducer( {}, {
 	[ JETPACK_CONNECTION_STATUS_REQUEST_SUCCESS ]: createRequestReducer( false )
 } );
 
+/**
+ * `Reducer` function which handles request/response actions
+ * concerning Jetpack user connection data
+ *
+ * @param  {Array}  state  Current state
+ * @param  {Object} action action
+ * @return {Array}         Updated state
+ */
+export const dataItems = createReducer( {}, {
+	[ JETPACK_USER_CONNECTION_DATA_RECEIVE ]: ( state, { siteId, data } ) => Object.assign( {}, state, { [ siteId ]: data } )
```

Comment:
It would be nice to use `keyedReducer` [like we do here](https://github.com/Automattic/wp-calypso/blob/2f73cb1c0342253c7cc442a271ea2951cd943b84/client/state/activity-log/restore/reducer.js#L43-L47).